tools/clang/unittests/HLSLTestLib/FileCheckerTest.cpp

```diff
@@ -783,8 +783,15 @@ FileRunCommandPart::RunD3DReflect(dxc::DxcDllSupport &DllSupport,
   IFT(DllSupport.CreateInstance(CLSID_DxcLibrary, &pLibrary));
   IFT(DllSupport.CreateInstance(CLSID_DxcAssembler, &pAssembler));
 
-  IFT(pLibrary->CreateBlobWithEncodingFromPinned(
-      (LPCVOID)Prior->StdOut.c_str(), Prior->StdOut.size(), CP_UTF8, &pSource));
+  // It would be better to properly honour %s for input, but for now, if prior
+  // output is empty, use the file as input.
```

Comment:
I'd like to understand why this was necessary

Reply:
It was necessary to input raw IR.  Normally, `%D3DReflect` follows a compilation (`%dxc`) or link (`%dxl`) where the resulting IR is dumped to the `Prior->StdOut`.  But if I want to start with IR, there is no prior stage dumping this IR to `Prior->StdOut`.  Ideally, I might do `%dxa` but that doesn't exist here, so I did this as a workaround for the problem.  The right fix would be to use the file if `%s` was specified, otherwise use the `Prior->StdOut`.  But to make that change, I'd have to change a whole lot more tests to remove the `%s` from `%D3DReflect %s` in their RUN lines.  I was thinking of doing that in a follow-up change.

Reply:
Filed #6271 to track.